src/ts/components/ads/AdControlBar.ts

```diff
@@ -0,0 +1,120 @@
+import { ControlBar, ControlBarConfig } from '../ControlBar';
+import { Container, ContainerConfig } from '../Container';
+import { PlayerAPI } from 'bitmovin-player';
+import { UIInstanceManager } from '../../UIManager';
+import { i18n } from '../../localization/i18n';
+import { SeekBar } from '../seekbar/SeekBar';
+
+
+export interface AdControlBarConfig extends ControlBarConfig {}
+
+/**
+ * Contains player control components displayed during ad playback,
+ * e.g., play toggle button, seek bar, volume control, fullscreen toggle button.
+ * 
+ * Usage: Pass one or more {@link Container} components via the `components` array.
+ * - Containers containing a {@link SeekBar} (directly or nested) are always shown.
+ * - Other containers are hidden/shown when controls hide.
+ *
+ * @example
+ * ```typescript
+ * new AdControlBar({
+ *   components: [
+ *     new Container({
+ *       components: [
+ *         new AdCounterLabel(),
+ *         new SeekBar({ label: new SeekBarLabel() }),
+ *         new PlaybackTimeLabel({ timeLabelMode: PlaybackTimeLabelMode.RemainingTime }),
+ *       ],
+ *       cssClasses: ['ad-controlbar-top'],
+ *     }),
+ *     new Container({
+ *       components: [
+ *         new PlaybackToggleButton(),
+ *         new VolumeToggleButton(),
+ *         new Spacer(),
+ *         new FullscreenToggleButton(),
+ *       ],
+ *       cssClasses: ['ad-controlbar-bottom'],
+ *     }),
+ *   ],
+ * })
+ * ```
+ *
+ * @category Components
+ */
+export class AdControlBar extends ControlBar {
+  private containersToHide: Container<ContainerConfig>[] = [];
+  private containersToKeepVisible: Container<ContainerConfig>[] = [];
+
+  constructor(config: AdControlBarConfig) {
+    super({
+      ...config,
+      cssClasses: ['ad-controlbar'],
+    });
```

Comment:
Can we keep our current style of calling super in components: (and see comment below)


```suggestion
    super(config);
```

Reply:
Address in 75d263672ddb3d9313b927396e09d3156d252636